### PR TITLE
Created RecursiveUploadEnabledInterface

### PR DESCRIPTION
**Pull request by oscarotero**
According to https://github.com/deployphp/deployer/pull/891#issuecomment-262900319

Note that in NativeSsh, the download method is recursive too, to be able to download not only files but also directories.

| Q             | A
| ------------- | ---
| Bug fix?      | No
| New feature?  | Yes
| BC breaks?    | No
| Deprecations? | No
| Fixed tickets | N/A

### CODE REVIEW

File: src/Server/Remote/NativeSsh.php [hunk 1]
```diff
@@ -106,27 +107,39 @@ public function download($local, $remote)
...
 
-        $scpOptions = [];
-
         if ($serverConfig->getPort()) {
             $scpOptions[] = '-P ' . escapeshellarg($serverConfig->getPort());
         }
 
         if ($serverConfig->getPrivateKey()) {
-            $sshOptions[] = '-i ' . escapeshellarg($serverConfig->getPrivateKey());
+            $scpOptions[] = '-i ' . escapeshellarg($serverConfig->getPrivateKey());
```
Comment (oscarotero) [inline src/Server/Remote/NativeSsh.php]: This seemed a bug, so it was changed too

**Comment by elfet** [comment]
@oscarotero can you rebase, please?

**Comment by oscarotero** [comment]
Sorry, I'm not sure if I did it correctly.
If you want, I can close this and create a new clean pull request.

**Comment by elfet** [comment]
Yeah, something goes wrong)

**Pull request closed without merging**